### ISSUE TITLE
Incorrect empty input handling in .download_all()

### ISSUE THREAD

**Issue by Rim-chan** [sentinelsat/sentinelsat#377]
I am trying to download sentinel-2A images using google collab and sentinelsat and ran into an error.
Here is the code snippets I used:

```python
from sentinelsat.sentinel import SentinelAPI, read_geojson, geojson_to_wkt
from datetime import date

# connect to the API
api = SentinelAPI('user', 'password', 'https://scihub.copernicus.eu/dhus')

# search by polygon, time, and Hub query keywords
footprint = geojson_to_wkt(read_geojson('./Bizerte_BoundaryMap/smaller_region.geojson'))
products = api.query(footprint,
                     date = ('20190625', date(2019, 6, 25)),
                     platformname = 'Sentinel-2',
                     processinglevel = 'Level-2A',
                     cloudcoverpercentage = (0, 10))

# download all results from the search
api.download_all(products)
```
and I got the following error:
```
---------------------------------------------------------------------------
StopIteration                             Traceback (most recent call last)
<ipython-input-29-1da24c759dfe> in <module>()
      1 # download all results from the search
----> 2 api.download_all(products)

/usr/local/lib/python3.6/dist-packages/sentinelsat/sentinel.py in download_all(self, products, directory_path, max_attempts, checksum, n_concurrent_dl, lta_retry_delay)
    805 
    806         if len(failed_prods) == len(product_ids):
--> 807             raise next(iter(x.exception() for x in dl_tasks if x.exception()))
    808 
    809         return downloaded_prods, retrieval_scheduled, failed_prods

StopIteration: 
```

I have also tried to download the data via cmd commands. However, I ran into the following error:

```
ERROR: Error checking for conflicts.
Traceback (most recent call last):
  File "C:\Users\ASUS\Anaconda3\lib\site-packages\pip\_vendor\pkg_resources\__init__.py", line 3021, in _dep_map
    return self.__dep_map
  File "C:\Users\ASUS\Anaconda3\lib\site-packages\pip\_vendor\pkg_resources\__init__.py", line 2815, in __getattr__
    raise AttributeError(attr)
AttributeError: _DistInfoDistribution__dep_map

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\ASUS\Anaconda3\lib\site-packages\pip\_vendor\pkg_resources\__init__.py", line 3012, in _parsed_pkg_info
    return self._pkg_info
  File "C:\Users\ASUS\Anaconda3\lib\site-packages\pip\_vendor\pkg_resources\__init__.py", line 2815, in __getattr__
    raise AttributeError(attr)
AttributeError: _pkg_info

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\ASUS\Anaconda3\lib\site-packages\pip\_internal\commands\install.py", line 517, in _warn_about_conflicts
    package_set, _dep_info = check_install_conflicts(to_install)
  File "C:\Users\ASUS\Anaconda3\lib\site-packages\pip\_internal\operations\check.py", line 114, in check_install_conflicts
    package_set, _ = create_package_set_from_installed()
  File "C:\Users\ASUS\Anaconda3\lib\site-packages\pip\_internal\operations\check.py", line 53, in create_package_set_from_installed
    package_set[name] = PackageDetails(dist.version, dist.requires())
  File "C:\Users\ASUS\Anaconda3\lib\site-packages\pip\_vendor\pkg_resources\__init__.py", line 2736, in requires
    dm = self._dep_map
  File "C:\Users\ASUS\Anaconda3\lib\site-packages\pip\_vendor\pkg_resources\__init__.py", line 3023, in _dep_map
    self.__dep_map = self._compute_dependencies()
  File "C:\Users\ASUS\Anaconda3\lib\site-packages\pip\_vendor\pkg_resources\__init__.py", line 3032, in _compute_dependencies
    for req in self._parsed_pkg_info.get_all('Requires-Dist') or []:
  File "C:\Users\ASUS\Anaconda3\lib\site-packages\pip\_vendor\pkg_resources\__init__.py", line 3014, in _parsed_pkg_info
    metadata = self.get_metadata(self.PKG_INFO)
  File "C:\Users\ASUS\Anaconda3\lib\site-packages\pip\_vendor\pkg_resources\__init__.py", line 1420, in get_metadata
    value = self._get(path)
  File "C:\Users\ASUS\Anaconda3\lib\site-packages\pip\_vendor\pkg_resources\__init__.py", line 1616, in _get
    with open(path, 'rb') as stream:
FileNotFoundError: [Errno 2] No such file or directory: 'c:\\users\\asus\\anaconda3\\lib\\site-packages\\jmespath-0.9.5.dist-info\\METADATA'

C:\Users\ASUS> conda install sentinelsat
Collecting package metadata (current_repodata.json): done
Solving environment: failed with initial frozen solve. Retrying with flexible solve.
Collecting package metadata (repodata.json): done
Solving environment: failed with initial frozen solve. Retrying with flexible solve.

PackagesNotFoundError: The following packages are not available from current channels:

  - sentinelsat

Current channels:

  - https://repo.anaconda.com/pkgs/main/win-64
  - https://repo.anaconda.com/pkgs/main/noarch
  - https://repo.anaconda.com/pkgs/r/win-64
  - https://repo.anaconda.com/pkgs/r/noarch
  - https://repo.anaconda.com/pkgs/msys2/win-64
  - https://repo.anaconda.com/pkgs/msys2/noarch

To search for alternate channels that may provide the conda package you're
looking for, navigate to

    https://anaconda.org

and use the search bar at the top of the page.
```

**Comment by valgur** [comment]
Regarding the first problem, it's not clear what exactly went wrong there. The error handling needs some improvement, though, the `StopIteration` exception should not happen.
Please share the GeoJSON or WKT you are using in the query, if possible, so we can hopefully reproduce and debug the error.

As for the conda error, sentinelsat is simply not provided as a conda package. You will have to install it via `pip install sentinelsat`. If you get a pip error, then try updating pip first: `python -m pip install -U pip`.

**Comment by Rim-chan** [comment]
Thank you for the reply. How can I share the GeoJSON file ? 
I encountered the same problem when using pip install sentinelsat.

**Comment by valgur** [comment]
You can copy the output of the following here:
```python
from sentinelsat.sentinel import read_geojson, geojson_to_wkt
footprint = geojson_to_wkt(read_geojson('./Bizerte_BoundaryMap/smaller_region.geojson'))
print(footprint)
```

**Comment by Rim-chan** [comment]
I did that and got this error:
 File "<ipython-input-8-c93d799f1145>", line 10
    products = api.query(POLYGON((10.1486 37.1997 , 9.9481 37.1910 , 9.8712 37.1034,9.7696 37.1056,9.6625 37.0661,9.6295 37.0091,9.7394 36.9411,9.9097 36.9850,9.9646 36.9828,10.0690 37.0464,10.1486 37.1997)),
                                                ^
SyntaxError: invalid syntax

**Comment by valgur** [comment]
Uh, not sure what you tried to do there. Please just run the code snippet I shared in a Python console or in the Google Colab notebook and share the result.

**Comment by j08lue** [comment]
> not sure what you tried to do there

footprint WKT raw pasted into code, not as string.

**Comment by Rim-chan** [comment]
Then how?

**Comment by Rim-chan** [comment]
this is what I did @valgur 
`from sentinelsat.sentinel import SentinelAPI, read_geojson, geojson_to_wkt
from datetime import date


# connect to the API
api = SentinelAPI('Rym', 'thedevilwearsparada', 'https://scihub.copernicus.eu/dhus')

# search by polygon, time, and Hub query keywords
footprint = geojson_to_wkt(read_geojson('./Bizerte_BoundaryMap/smaller_region.geojson'))
products = api.query(footprint="POLYGON((10.1486 37.1997 , 9.9481 37.1910 , 9.8712 37.1034,9.7696 37.1056,9.6625 37.0661,9.6295 37.0091,9.7394 36.9411,9.9097 36.9850,9.9646 36.9828,10.0690 37.0464,10.1486 37.1997))",
                     date = ('20190625', date(2019, 6, 25)),
                     platformname = 'Sentinel-2',
                     processinglevel = 'Level-2A',
                     cloudcoverpercentage = (0,5)
                     )`

**Comment by agrisbrauns** [comment]
I think I have encountered a similar problem.

I used the following script to download Sentinel-2 tiles:

```
import os
from collections import OrderedDict
from sentinelsat import SentinelAPI
from datetime import date

api = SentinelAPI("XXXXXX", "XXXXXXX")

tiles = ["35VMD", "35VLD", "35VLE"]

query_kwargs = {
    "platformname": "Sentinel-2",
    "producttype": "S2MSI1C",
    "cloudcoverpercentage": (0, 100),
    'date': ("NOW-1DAY", "NOW")

}

products = OrderedDict()
for tile in tiles:
    kw = query_kwargs.copy()
    kw["tileid"] = tile 
    pp = api.query(**kw)
    products.update(pp)

api.download_all(products)
```

and it returns:

`Traceback (most recent call last):
  File "download_S2.py", line 36, in <module>
    api.download_all(products)
  File "C:\ProgramData\Anaconda3\lib\site-packages\sentinelsat\sentinel.py", line 807, in download_all
    raise next(iter(x.exception() for x in dl_tasks if x.exception()))
StopIteration`

But when I specify the date as  ("NOW-2DAYS", "NOW") it works. It is similar with HOUR(S) - for some specified time interval the api.download_all works for some it does not.

**Comment by Rim-chan** [comment]
Do I need to specify the tiles like this?  `tiles = ["35VMD", "35VLD", "35VLE"]` I can't use a GeoJSON file for that ?

**Comment by agrisB** [comment]
@Rim-chan you can use GeoJSON. I am specifying the tiles as my analysis is based on specific tiles. However, it does not change the fact that download_all() throws a "StopIteration". Currently, I can not understand whether it is due to the specified date or due to files that are (or are not) covered within the specified time frame.

**Comment by Squiccio** [comment]
I have the same problem but it happens only when products are not founded. If Sentinelsat finds product, with json or tile queries, it works. This is my error:
Found 0 products
Will download 0 products using 2 workers
Traceback (most recent call last):
  File "h:\anaconda3\envs\santinelsat_last\lib\runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "h:\anaconda3\envs\santinelsat_last\lib\runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "H:\Anaconda3\envs\Santinelsat_last\Scripts\sentinelsat.exe\__main__.py", line 7, in <module>
  File "h:\anaconda3\envs\santinelsat_last\lib\site-packages\click\core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "h:\anaconda3\envs\santinelsat_last\lib\site-packages\click\core.py", line 782, in main
    rv = self.invoke(ctx)
  File "h:\anaconda3\envs\santinelsat_last\lib\site-packages\click\core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "h:\anaconda3\envs\santinelsat_last\lib\site-packages\click\core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "h:\anaconda3\envs\santinelsat_last\lib\site-packages\sentinelsat\scripts\cli.py", line 228, in cli
    product_infos, triggered, failed_downloads = api.download_all(products, path)
  File "h:\anaconda3\envs\santinelsat_last\lib\site-packages\sentinelsat\sentinel.py", line 807, in download_all
    raise next(iter(x.exception() for x in dl_tasks if x.exception()))
StopIteration

Thanks

**Comment by agrisB** [comment]
@Squiccio Yes, the error seems to be related to the lack of products within the queried period of time or other filtered properties.

**Comment by valgur** [comment]
The bug indeed seems to be caused by simply handling empty input to `.download_all()` incorrectly. Should get fixed with #378, but I'll add a test for it as well just to be sure.
Thanks for the extra info @Squiccio and @agrisB!
